code/game/machinery/telecomms/telecomunications.dm

```diff
@@ -115,6 +134,40 @@ GLOBAL_LIST_EMPTY(telecomms_list)
 					links |= T
 					T.links |= src
 
+/obj/machinery/telecomms/proc/update_network()
+	if(!network || network == "NULL")
+		return
+	var/new_network_id = NETWORK_NAME_COMBINE(__NETWORK_SERVER, network) // should result in something like SERVER.TCOMMSAT
+	var/area/A = get_area(src)
+	if(A)
+		if(!A.network_root_id)
+			log_telecomms("Area '[A.name]([REF(A)])' has no network network_root_id, force assigning in object [src]([REF(src)])")
+			SSnetworks.lookup_area_root_id(A)
+			new_network_id = NETWORK_NAME_COMBINE(A.network_root_id, new_network_id) // should result in something like SS13.SERVER.TCOMMSAT
+		else
+			log_telecomms("Created [src]([REF(src)] in nullspace, assuming network to be in station")
+			new_network_id = NETWORK_NAME_COMBINE(STATION_NETWORK_ROOT, new_network_id) // should result in something like SS13.SERVER.TCOMMSAT
+	new_network_id = simple_network_name_fix(new_network_id) // make sure the network name is valid
+	var/datum/ntnet/new_network = SSnetworks.create_network_simple(new_network_id)
+	new_network.move_interface(GetComponent(/datum/component/ntnet_interface), new_network_id, network_id)
+	network_id = new_network_id
+
+/obj/machinery/telecomms/proc/ntnet_receive(datum/source, datum/netdata/data)
+
+	//Check radio signal jamming
+	if(is_jammed(JAMMER_PROTECTION_WIRELESS) || machine_stat & (BROKEN|NOPOWER|MAINT|EMPED))
+		return
+
+	switch(data.data["type"])
+		if("ping") // we respond to the ping with our status
```

Comment:
this should be a macro.

Reply:
not sure what do you want me to make a macro

Reply:
`#define DATA_TYPE_PING "ping"`

Reply:
done